### PR TITLE
plugin dev for snapshot have tags

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/aws/ec2/ebsSnapShotHasTags.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'EBS Snapshot Has Tags',
+    category: 'EC2',
+    domain: 'Compute',
+    description: 'Ensure that EBS snapshots have tags',
```
Comment (alphadev4) [inline plugins/aws/ec2/ebsSnapShotHasTags.js:8]:
```suggestion
    description: 'Ensure that EBS snapshots have tags associated.',
```

File: plugins/aws/ec2/ebsSnapShotHasTags.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            if (!describeSnapshots) return rcb();
+
+            if (describeSnapshots.err || !describeSnapshots.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query for EBS Snapshots: ${helpers.addError(describeSnapshots)}`, region);
+                return rcb();
+            }
+
+            if (!describeSnapshots.data.length) {
+                helpers.addResult(results, 0, 'No EBS snapshots present', region);
```
Comment (alphadev4) [inline plugins/aws/ec2/ebsSnapShotHasTags.js:33]:
```suggestion
                helpers.addResult(results, 0, 'No EBS snapshots found', region);
```

File: plugins/aws/ec2/ebsSnapShotHasTags.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                return rcb();
+            }
+            for (let snapshot of describeSnapshots.data){
+                var resourceARN = `arn:${awsOrGov}:${region}:${snapshot.OwnerId}:snapshot/${snapshot.SnapshotId}`;
+
+                if (!snapshot.Tags || !snapshot.Tags.length) {
+                    helpers.addResult(results, 2, 'EBS Snapshot has no Tags associated.', region, resourceARN);
+                } else {
+                    helpers.addResult(results, 0, 'EBS Snapshot have tags.', region, resourceARN);
```
Comment (alphadev4) [inline plugins/aws/ec2/ebsSnapShotHasTags.js:42]:
```suggestion
                    helpers.addResult(results, 0, 'EBS Snapshot has tags.', region, resourceARN);
```

File: plugins/aws/ec2/ebsSnapShotHasTags.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+            if (!describeSnapshots.data.length) {
+                helpers.addResult(results, 0, 'No EBS snapshots present', region);
+                return rcb();
+            }
+            for (let snapshot of describeSnapshots.data){
+                var resourceARN = `arn:${awsOrGov}:${region}:${snapshot.OwnerId}:snapshot/${snapshot.SnapshotId}`;
+
+                if (!snapshot.Tags || !snapshot.Tags.length) {
+                    helpers.addResult(results, 2, 'EBS Snapshot has no Tags associated.', region, resourceARN);
```
Comment (alphadev4) [inline plugins/aws/ec2/ebsSnapShotHasTags.js:40]:
```suggestion
                    helpers.addResult(results, 2, 'EBS Snapshot does not have tags', region, resourceARN);
```

File: plugins/aws/ec2/ebsSnapshotHasTags.spec.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    };
+};
+
+describe('ebsSnapShotHasTags', function () {
+    describe('run', function () {
+        it('should PASS if EBS snapshot has Tags', function (done) {
+            const cache = createCache([describeSnapshots[1]]);
+            ebsSnapShotHasTags.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
```
[the remark this answers is not in the document]
Comment (alphadev4) [reply plugins/aws/ec2/ebsSnapshotHasTags.spec.js:76]: Make sure to include message check in spec files

**Review by alphadev4 — APPROVED** [review]

File: exports.js [hunk 6]
```diff
@@ -204,6 +204,7 @@ module.exports = {
         'ec2HasTags'                    : require(__dirname + '/plugins/aws/ec2/ec2HasTags.js'),
         'securityGroupHasTags'          : require(__dirname + '/plugins/aws/ec2/securityGroupsHasTags.js'),
         'vpcHasTags'                    : require(__dirname + '/plugins/aws/ec2/vpcHasTags'),
+        'ebsSnapShotHasTags'            : require(__dirname + '/plugins/aws/ec2/ebsSnapShotHasTags'),
```
Comment (AkhtarAmir) [inline exports.js:207]:
```suggestion
        'ebsSnapshotHasTags'            : require(__dirname + '/plugins/aws/ec2/ebsSnapshotHasTags'),
```

File: plugins/aws/ec2/ebsSnapShotHasTags.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+                helpers.addResult(results, 3,
+                    `Unable to query for EBS Snapshots: ${helpers.addError(describeSnapshots)}`, region);
+                return rcb();
+            }
+
+            if (!describeSnapshots.data.length) {
+                helpers.addResult(results, 0, 'No EBS snapshots found', region);
+                return rcb();
+            }
+            for (let snapshot of describeSnapshots.data){
```
Comment (AkhtarAmir) [inline plugins/aws/ec2/ebsSnapShotHasTags.js:36]:
```suggestion
            }

            for (let snapshot of describeSnapshots.data){
```

File: plugins/aws/ec2/ebsSnapShotHasTags.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+                if (!snapshot.OwnerId || !snapshot.SnapshotId) continue;
+
+                var resourceARN = `arn:${awsOrGov}:${region}:${snapshot.OwnerId}:snapshot/${snapshot.SnapshotId}`;
+
+                if (!snapshot.Tags || !snapshot.Tags.length) {
+                    helpers.addResult(results, 2, 'EBS Snapshot has no Tags associated', region, resourceARN);
```
Comment (AkhtarAmir) [inline plugins/aws/ec2/ebsSnapShotHasTags.js:42]:
```suggestion
                    helpers.addResult(results, 2, 'EBS Snapshot does not have tags', region, resourceARN);
```

File: plugins/aws/ec2/ebsSnapShotHasTags.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+                } else {
+                    helpers.addResult(results, 0, 'EBS Snapshot have tags', region, resourceARN);
```
Comment (AkhtarAmir) [inline plugins/aws/ec2/ebsSnapShotHasTags.js:44]:
```suggestion
                    helpers.addResult(results, 0, 'EBS Snapshot has tags', region, resourceARN);
```

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**